### PR TITLE
* Fix modal for real

### PR DESCRIPTION
**Pull request by vseguin**
Using the good key help 🤦‍♂️ 

@sroy3  we still have a problem though... if we have multiple modals opened, they _all_ close when pressing X. Pressing escape works correctly though. I'm not sure how to fix this one...

EDIT: I have a tentative fix in there, feel free to review properly.

### CODE REVIEW

File: src/components/modal/ModalHeader.tsx [hunk 1]
```diff
@@ -18,9 +21,14 @@ export interface IModalHeaderDispatchProps {
 export interface IModalHeaderProps extends IModalHeaderOwnProps, IModalHeaderStateProps, IModalHeaderDispatchProps {}
 
 export class ModalHeader extends React.Component<IModalHeaderProps, {}> {
+    private canClose: boolean = true;
+
+    componentDidUpdate() {
+        this.canClose = _.isUndefined(this.props.lastOpened) ? true : this.props.lastOpened;
```
Comment (GermainBergeron) [inline src/components/modal/ModalHeader.tsx]: What do you think of this syntax?: `this.canClose = _.result(this.props, 'lastOpened', true);`

Comment (vseguin) [reply src/components/modal/ModalHeader.tsx]: I like it! Will change.

File: src/components/modal/ModalHeaderConnected.tsx [hunk 2]
```diff
@@ -1,9 +1,20 @@
 import {connect} from 'react-redux';
+import * as _ from 'underscore';
+import {IReactVaporState} from '../../ReactVapor';
 import {IDispatch, ReduxUtils} from '../../utils/ReduxUtils';
 import {closeModal} from './ModalActions';
-import {IModalHeaderDispatchProps, IModalHeaderOwnProps, IModalHeaderProps, ModalHeader} from './ModalHeader';
+import {
+    IModalHeaderDispatchProps, IModalHeaderOwnProps, IModalHeaderProps, IModalHeaderStateProps,
+    ModalHeader,
```
Comment (sroy3) [inline src/components/modal/ModalHeaderConnected.tsx]: Optimize imports

**Comment by sroy3** [comment]
Demo: https://coveo.github.io/react-vapor/fix-modal-for-real/

**Comment by sroy3** [comment]
It does not seem to work on the "Modal Composite Connected with another modal inside" example. It does not work using the X and the close button of the second modal.

**Comment by vseguin** [comment]
@sroy3 Was it working before? I never saw it working actually...

File: src/components/modal/ModalHeaderConnected.tsx [hunk 3]
```diff
@@ -1,9 +1,20 @@
...
+    IModalHeaderDispatchProps, IModalHeaderOwnProps, IModalHeaderProps, IModalHeaderStateProps,
+    ModalHeader,
+} from './ModalHeader';
 
-const mapStateToProps = () => ({});
+const mapStateToProps = (state: IReactVaporState, ownProps: IModalHeaderOwnProps): IModalHeaderStateProps => {
+    const lastOpenedModal = _.last(state.openModals);
+
+    return {
+        lastOpened: ownProps.id === lastOpenedModal,
```
Comment (MichelML) [inline src/components/modal/ModalHeaderConnected.tsx]: you could just name this `canClose` if this is the only condition for it to close, or just use `this.props.lastOpened`   
  
```  
if (this.canClose && this.props.onClose) {  
```

File: src/components/modal/tests/ModalHeaderConnected.spec.tsx [hunk 4]
```diff
@@ -31,6 +31,7 @@ describe('ModalHeader', () => {
                     <ModalHeaderConnected
                         id={id}
                         title={title}
+                        lastOpened={true}
```
Comment (MichelML) [inline src/components/modal/tests/ModalHeaderConnected.spec.tsx]: if this is a boolean prop in react, only putting `lastOpened` means the same as `lastOpened={true}`

**Comment by sroy3** [comment]
You have 3 tests failing

**Pull request merged**